pkg/specgen/generate/kube/kube.go

```diff
@@ -126,13 +125,29 @@ func ToSpecGen(ctx context.Context, containerYAML v1.Container, iid string, newI
 			s.StopSignal = &stopSignal
 		}
 	}
-	if len(containerYAML.Command) != 0 {
-		s.Command = containerYAML.Command
-	}
+
 	// doc https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes
-	if len(containerYAML.Args) != 0 {
-		s.Command = append(s.Command, containerYAML.Args...)
+
+	if len(containerYAML.Command) == 0 && len(containerYAML.Args) == 0 {
+		s.Entrypoint = imageData.Config.Entrypoint
+		s.Command = imageData.Config.Cmd
+	}
+
+	if len(containerYAML.Command) != 0 && len(containerYAML.Args) == 0 {
+		s.Entrypoint = containerYAML.Command
+		s.Command = nil
 	}
+
+	if len(containerYAML.Command) == 0 && len(containerYAML.Args) != 0 {
+		s.Entrypoint = imageData.Config.Entrypoint
+		s.Command = containerYAML.Args
+	}
+
+	if len(containerYAML.Command) != 0 && len(containerYAML.Args) != 0 {
+		s.Entrypoint = containerYAML.Command
+		s.Command = containerYAML.Args
+	}
```

Comment:
Hi! can we boil this down to a simple "set entrypoint/command from `imageData` and override from `containerYAML` when set" ?  I haven't seen this PR before I started fixing this and this would be my patch of this file: https://github.com/containers/podman/compare/master...bziemons:patch-kube-play-8428.patch

I also want to emphasize that the `if imageData != nil && imageData.Config != nil {` above is probably there for a reason and accessing `imageData.Config` without it is probably not a good idea.

Reply:
agree with @bziemons, line: 132,142 need the the imageData != nil && imageData.Config != nil

Reply:
> Hi! can we boil this down to a simple "set entrypoint/command from `imageData` and override from `containerYAML` when set" ? I haven't seen this PR before I started fixing this and this would be my patch of this file: https://github.com/containers/podman/compare/master...bziemons:patch-kube-play-8428.patch

Yes, we can do that. It is functionally equivalent and less code, but it lacks the clarity in mirroring the k8s specs that I wished to express. I feel that your proposition warrants some line of comments to establish that link, but apart from that, I wouldn't complain.

> I also want to emphasize that the `if imageData != nil && imageData.Config != nil {` above is probably there for a reason and accessing `imageData.Config` without it is probably not a good idea.

I fear I lost the context here. Are you referring to L115 before setting `s.WorkDir = imageData.Config.WorkingDir`? I did not notice that, but I supsect it is not necessary, because `imageData` is populated with `imageData, err := newImage.Inspect(ctx)` at L110. Any failure to get the `imageData` returns there.

In addition, I cannot imagine how to start any container without having both `imageData` and `imageData.Config`. Accordingly, the corresponding check, if it really is necessary, is misplaced when setting single values and should be performed at the "global" level somewhar at `libpod/image/image.go`, `Inspect` or when examining the return value  of it's invocation at L110.

Reply:
>> Hi! can we boil this down to a simple "set entrypoint/command from `imageData` and override from `containerYAML` when set" ? I haven't seen this PR before I started fixing this and this would be my patch of this file: https://github.com/containers/podman/compare/master...bziemons:patch-kube-play-8428.patch

> I also want to emphasize that the `if imageData != nil && imageData.Config != nil {` above is probably there for a reason and accessing `imageData.Config` without it is probably not a good idea.

Ok, now I get it. That part, if it is really necessary, about which I am yet not sure, should error much higher in the chain.

Reply:
I agree with you and the if should have gotten a comment from its writer about why it is needed. Even if it isn't needed it wouldn't make sense to make the code more confusing about it (I know that wasn't your intention ;) ). Thanks for accepting my feedback though